ClientApp/src/components/Groceries.js

```diff
@@ -0,0 +1,65 @@
+﻿import React, { Component } from 'react';
+
+export class Groceries extends Component {
+    constructor() {
+        super();
+        this.state = {
+            groceries: []
+        }
+    }
+
+    handleSubmit = (e) => {
+        e.preventDefault();
+        let groceries = this.state.groceries;
+        let storeName = this.refs.storeName.value;
+        let itemName = this.refs.itemName.value;
+        let itemPrice = this.refs.itemPrice.value;
+
+        let newGroceries = {
+            "storeName" : storeName,
+            "itemName" : itemName,
+            "itemPrice" : itemPrice,
+        }
+        groceries.push(newGroceries);
+
+        this.setState({
+            groceries : groceries
+        })
+
+        this.refs.groceriesForm.reset();
+    }
+
+    render(){
+        let groceries = this.state.groceries;
+        return(
+            <div>
+                <form ref="groceriesForm">
+                <label>Store Name</label>
+                <input type="text" ref="storeName"/>
```

Comment:
Here you should use `<Input>` tag provided by `reactstrap`  package. It applies `bootstrap` theme that is global to the project. Same goes for `<button>` tag. You should use `<Button>` instead.